From 0daf9017b8db1c04a8407976036d9f84dddd8c1e Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@st.com>
Date: Fri, 22 Apr 2022 14:57:27 +0200
Subject: [PATCH 0411/1695] dts: stm32: update HSEM node in stm32mp251.dtsi

Update HSEM node definition in stm32mp251.dtsi with its compatible,
interrupt line and clock.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 12f406b800d8deae520d237e811b5d8e8e902350 dts: stm32: update HSEM node in stm32mp251.dtsi]
Signed-off-by: Gatien Chevallier <gatien.chevallier@st.com>
Change-Id: Iad6549861aed32821a9d05bdf04c3cc6b46f31b6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/249376
Reviewed-by: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 56ce67eb6..92573e66d 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -709,8 +709,10 @@
 		};
 
 		hsem: hwspinlock@46240000 {
+			compatible = "st,stm32mp25-hsem";
 			reg = <0x46240000 0x400>;
-			status = "disabled";
+			interrupts = <GIC_SPI 200 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&rcc CK_BUS_HSEM>;
 		};
 
 		ipcc2: mailbox@46250000 {
-- 
2.47.2

